homeassistant/components/light/flux_led.py

```diff
@@ -253,24 +247,23 @@ def turn_on(self, **kwargs):
         if brightness is None:
             brightness = self.brightness
 
-        # Preserve color on brightness/white level change
-        if rgb is None:
-            rgb = self._bulb.getRgb()
-
-        if white is None and self._mode == MODE_RGBW:
-            white = self.white_value
+        if hs_color and brightness:
+            self._color = color_util.color_hsv_to_RGB(hs_color[0], hs_color[1],
+                                                      brightness / 255 * 100)
 
         # handle W only mode (use brightness instead of white value)
         if self._mode == MODE_WHITE:
             self._bulb.setRgbw(0, 0, 0, w=brightness)
 
         # handle RGBW mode
         elif self._mode == MODE_RGBW:
-            self._bulb.setRgbw(*tuple(rgb), w=white, brightness=brightness)
-
+            if white is None:
+                self._bulb.setRgbw(*tuple(self._color))
```

Comment:
When setting `brightness` from an off state, `hs_color` will be None, and this will not affect the brightness, as that happens on lines https://github.com/home-assistant/home-assistant/blob/87f66fd5fe614c12677f6b95df9d13e16242353f/homeassistant/components/light/flux_led.py#L250-L252

Reply:
I implemented it now, so when only brightness is given, it uses the saved color and calculates it with the given brightness.